parlai/agents/rag/retrievers.py

```diff
@@ -640,6 +640,13 @@ def index_retrieve(
             vectors = vectors[:, :, :-1]
         # recompute exact FAISS scores
         scores = torch.bmm(query.unsqueeze(1), vectors.transpose(1, 2)).squeeze(1)
+        if torch.isnan(scores).sum().item():
+            logging.error(
+                '\n[ Document scores are NaN; please look into the built index. ]\n'
+                '[ If using a compressed index, try building an exact index: ]\n'
+                '[ $ python index_dense_embeddings --indexer-type exact... ]'
+            )
+            scores.fill_(1)
```

Comment:
I'm curious why `1` is the error value?

Reply:
i picked an arbitrary number so that the scores play nicely with the rest of the system

the scores are literally a tensor of `nan`s:

```
ipdb> scores
tensor([[nan, nan, nan, nan, nan]], device='cuda:0', dtype=torch.float16)
```

so needed to pick something that worked